cmake/PostgreSQLExtension.cmake

```diff
@@ -255,6 +255,16 @@ function(add_postgresql_extension NAME)
                 PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
     endif()
 
+    # Check that the extension has no conflicting symbols with the Postgres binary
+    # otherwise the linker might use the symbols from Postgres
+    find_package(Python COMPONENTS Interpreter REQUIRED)
+    execute_process(COMMAND dirname ${PG_CTL} OUTPUT_VARIABLE POSTGRES_BINARY)
+    get_filename_component(POSTGRES_BINARY ${PG_CTL}/../postgres ABSOLUTE)
```

Comment:
couldn't find a nicer way of obtaining the postgres binary path

Reply:
FindPostgreSQL.cmake defines `PG_BINARY`